coreos-installer

```diff
@@ -378,6 +378,43 @@ write_networking_opts() {
     echo "set ignition_network_kcmdline=\"$(cat /tmp/networking_opts)\"" >> /mnt/boot_partition/ignition.firstboot
 }
 
+#########################################################
+# Helper to persist ignition platform id to initramfs first boot
+#########################################################
+write_platform_id() {
+    local tmp_platform_id="/tmp/platform_id"
+    local proc_cmdline="/proc/cmdline"
+
+    if [ ! -f ${tmp_platform_id} ]; then
+        log "Not overwriting ignition platform id, no platform id provided"
+        return
+    else
+        local platform_id="$(cat ${tmp_platform_id})"
+    fi
+
+
+    log "Overwriting ignition platform id"
+    # check for the boot partition
+    mkdir -p /mnt/boot_partition
+    mount_boot_partition /mnt/boot_partition
+    trap 'umount /mnt/boot_partition; trap - RETURN' RETURN
+    local conf_files=$(ls /mnt/boot_partition/loader/entries/ | grep 'conf')
+
+    for f in $conf_files
+    do
+        # Update boot loader entry first
+        local conf_files_path="/mnt/boot_partition/loader/entries/$f"
+        if grep -q 'ignition.platform.id' $conf_files_path; then
+            sed -i "s/ignition.platform.id=.* /ignition.platform.id=${platform_id} /" $conf_files_path
+        else
+            echo " ignition.platform.id=${platform_id}" >> $conf_files_path
```

Comment:
does appending to the file like this work?

Reply:
This might cause `echo: input/ouput error`, I will make sure in the morning

Reply:
What I'm saying is that it looks like the BLS configs can't simply be appended to. Here is the file I see:

```
# cat /mnt/boot_partition/loader/entries/ostree-1-fedora-coreos.conf
initrd /ostree/fedora-coreos-338b4a3ef2f647059ff49fcb9288f743760ae2c4866caaef852ce00c6e828fb7/initramfs-5.2.9-200.fc30.x86_64.img
version 1
options root=/dev/disk/by-label/root rootflags=defaults,prjquota rw $ignition_firstboot mitigations=auto,nosmt console=tty0 console=ttyS0,115200n8 ignition.platform.id=qemu ostree=/ostree/boot.10
linux /ostree/fedora-coreos-338b4a3ef2f647059ff49fcb9288f743760ae2c4866caaef852ce00c6e828fb7/vmlinuz-5.2.9-200.fc30.x86_64
title Fedora 30.20190904.dev.0 (CoreOS preview) (ostree)
```

If somehow `ignition.platform.id=qemu` weren't in there and we ended up doing `echo " ignition.platform.id=${platform_id}" >> $conf_files_path` you'd end up with `ignition.platform.id=foo` on its own line at the bottom of the file. That's not what we want.

Am I missing something?

Reply:
It needs to be appended to the kernel argument section `options`, right? I was mistakenly thinking the whole *.conf file only contains kernel args.

Reply:
> It needs to be appended to the kernel argument section `options`, right?

right